flows/engine/engine.go

```diff
@@ -143,24 +139,30 @@ func resumeNode(run flows.FlowRun, node flows.Node, step flows.Step, event flows
 		return noDestination, step, nil
 	}
 
-	err := wait.End(run, step, event)
+	// try to resume our wait with the first caller event
+	err := wait.End(run, step, callerEvents[0])
 	if err != nil {
 		run.AddError(step, err)
 		run.Exit(flows.StatusErrored)
 		return noDestination, step, nil
 	}
 
+	// we can now apply the caller events
+	for _, event := range callerEvents {
+		run.ApplyEvent(step, event)
+	}
```

Comment:
I don't disagree, but mostly interested in maybe documenting here why we do the wait and apply in this order.. perhaps having a longer comment here would help. I imagine there is a good reason around flow exits and such..